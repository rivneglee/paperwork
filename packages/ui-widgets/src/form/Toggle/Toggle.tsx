import React, { FunctionComponent, ReactElement, SyntheticEvent } from 'react';
import ReactToggle from 'react-toggle';
import 'react-toggle/style.css';

import { FieldGroup } from '../FieldGroup';

interface Props {
  checked: boolean;
  onChange?: (e: SyntheticEvent) => void;
  label?: ReactElement<any> | string;
  disabled?: boolean;
}

const Toggle: FunctionComponent<Props> = ({
  checked,
  onChange,
  label,
  disabled = false,
}) => (
  <FieldGroup label={label}>
    <ReactToggle
      className="pw-toggle"
      disabled={disabled}
      checked={checked}
      icons={{
        unchecked: null,
      }}
      onChange={onChange}
    />
  </FieldGroup>
);

export default Toggle;
