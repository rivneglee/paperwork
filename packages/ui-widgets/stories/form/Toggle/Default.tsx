import React, { useState } from 'react';

import { Toggle } from '../../../src';

export default () => {
  const [checked, setChecked] = useState(false);
  return (
    <div style={{ background: '#ebeef1', padding: '2.4rem' }}>
      <Toggle label="Enable feature" checked={checked} onChange={(e: any) => setChecked(e.target.checked)} />
      <Toggle disabled={true} label="Disabled" checked={true} />
    </div>
  );
};
