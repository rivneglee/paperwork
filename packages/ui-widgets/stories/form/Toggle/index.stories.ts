import { storiesOf } from '@storybook/react';

import Default from './Default';

storiesOf('Toggle', module)
  .add('default', Default);
