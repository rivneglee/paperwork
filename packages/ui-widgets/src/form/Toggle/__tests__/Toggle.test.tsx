import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import Toggle from '../Toggle';

describe('Toggle', () => {
  it('should render toggle when checked', () => {
    const wrapper: ShallowWrapper = shallow(<Toggle checked={true} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render toggle when unchecked', () => {
    const wrapper: ShallowWrapper = shallow(<Toggle checked={false} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render toggle when disabled', () => {
    const wrapper: ShallowWrapper = shallow(<Toggle checked={true} disabled />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
