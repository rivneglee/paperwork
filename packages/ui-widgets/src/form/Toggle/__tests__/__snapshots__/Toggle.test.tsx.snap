// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should render toggle when checked 1`] = `
<FieldGroup>
  <Toggle
    checked={true}
    className="pw-toggle"
    disabled={false}
    icons={
      Object {
        "unchecked": null,
      }
    }
  />
</FieldGroup>
`;

exports[`Toggle should render toggle when disabled 1`] = `
<FieldGroup>
  <Toggle
    checked={true}
    className="pw-toggle"
    disabled={true}
    icons={
      Object {
        "unchecked": null,
      }
    }
  />
</FieldGroup>
`;

exports[`Toggle should render toggle when unchecked 1`] = `
<FieldGroup>
  <Toggle
    checked={false}
    className="pw-toggle"
    disabled={false}
    icons={
      Object {
        "unchecked": null,
      }
    }
  />
</FieldGroup>
`;
